tests/test_postgres_incremental_replication.py

```diff
@@ -249,369 +343,449 @@ def get_properties():
     def test_run(self):
         conn_id = connections.ensure_connection(self)
 
-        # run in check mode
+        # run in check mode and verify exit codes
         check_job_name = runner.run_check_mode(self, conn_id)
-        # verify check  exit codes
         exit_status = menagerie.get_exit_status(conn_id, check_job_name)
         menagerie.verify_check_exit_status(self, exit_status, check_job_name)
 
-        # verify the tap discovered the right streams
-        found_catalogs = [fc for fc
-                          in menagerie.get_catalogs(conn_id)
-                          if fc['tap_stream_id'] in self.expected_check_streams()]
-
+        # verify basics of discovery are consistent with expectations...
 
-        self.assertGreaterEqual(len(found_catalogs),
-                                1,
-                                msg="unable to locate schemas for connection {}".format(conn_id))
+        # verify discovery produced (at least) 1 expected catalog
+        found_catalogs = [found_catalog for found_catalog in menagerie.get_catalogs(conn_id)
+                          if found_catalog['tap_stream_id'] in self.expected_check_streams()]
+        self.assertGreaterEqual(len(found_catalogs), 1)
 
-        found_catalog_names = set(map(lambda c: c['tap_stream_id'], found_catalogs))
-        diff = self.expected_check_streams().symmetric_difference(found_catalog_names)
-        self.assertEqual(len(diff), 0, msg="discovered schemas do not match: {}".format(diff))
+        # verify the tap discovered the expected streams
+        found_catalog_names = {catalog['tap_stream_id'] for catalog in found_catalogs}
+        self.assertSetEqual(self.expected_check_streams(), found_catalog_names)
 
         # verify that persisted streams have the correct properties
         test_catalog = found_catalogs[0]
         self.assertEqual(test_table_name, test_catalog['stream_name'])
-
         print("discovered streams are correct")
 
-        print('checking discoverd metadata for public-postgres_full_table_test...')
-        md = menagerie.get_annotated_schema(conn_id, test_catalog['stream_id'])['metadata']
-
-        self.assertEqual(
```

Comment:
Is this tested in the discovery test now?

Reply:
Yes, we check the metadata explicitly in the discovery test now. I left in the assertions above because I thought it made sense to.